src/actions/write/tick.ts

```diff
@@ -0,0 +1,78 @@
+import { DEFAULT_UNDERNAME_COUNT, SECONDS_IN_A_YEAR } from '../../constants';
+import { ContractResult, IOState } from '../../types';
+
+declare const SmartWeave: any;
+
+// Removes gateway from the gateway address registry after the leave period completes
+export const tick = async (state: IOState): Promise<ContractResult> => {
+  const { records, auctions, reserved } = state;
+
+  // TODO: do we give the caller a tip?
+
+  // remove any expired records
+  const activeRecords = Object.keys(records).reduce((current, key) => {
+    const record = records[key];
+    if (
+      (record.type === 'lease' &&
+        +record.endTimestamp > +SmartWeave.block.timestamp) ||
```

Comment:
should also check grace period, eg `+record.endTimestamp > +SmartWeave.block.timestamp + SECONDS_IN_GRACE_PERIOD`